test/helpers/helpers.go

```diff
@@ -428,31 +447,46 @@ func RestartMarketplace(client test.FrameworkClient, namespace string) error {
 	if err != nil {
 		return err
 	}
-	replicas := marketplace.Spec.Replicas
+	initialReplicas := marketplace.Spec.Replicas
 	zero := int32(0)
 	marketplace.Spec.Replicas = &zero
 	err = client.Update(context.TODO(), marketplace)
 	if err != nil {
 		return err
 	}
 	// Wait for deployment to scale down
-	err = WaitForDeploymentScaled(client, "marketplace-operator", namespace, zero)
+	err = ScaleMarketplace(client, namespace, int32(0))
```

Comment:
You need to wait for it scale down. If the scale down and up happens quickly, the net result will be a no-op as in no action will be taken.

Reply:
The `ScaleMarketplace()` method calls `WaitForDeploymentScaled()`, so it will wait.